### PR TITLE
Don't use https when building URLs if site does not support it

### PR DESCRIPTION
**Pull request by scottcwilson**
Removes https for localhost installs

### CODE REVIEW

**Comment by scottcwilson** [comment]
hmm... that logic doesn't seem to work on my installation.  But I will rework my fix so that the checks are in place function correctly with MAMP.   Right now when I run curlGetUrl, I get 

    [errmsg] => Unknown SSL protocol error in connection to localhost:-9847

**Comment by scottcwilson** [comment]
The code you are talking about affects whether the zc_install URL is SSL.  The code I'm talking about is building the URLs for the configure.php files.  That's where my change is.

**Comment by scottcwilson** [comment]
I updated the logic so it uses the ping response (which drives whether the page is loaded via SSL or not) in the decision as to whether to create URLs in configure.php with https:// or with http://.

File: zc_install/includes/functions/general.php [hunk 1]
```diff
@@ -199,8 +199,13 @@ function getDetectedURIs($adminDir = 'admin') {
...
     $catalogHttpUrl = 'http://' . $httpServer . '/' . zen_parse_url($url, 'path', true);
-    $catalogHttpsServer = 'https://' . $httpServer;
-    $catalogHttpsUrl = 'https://' . $httpServer . '/' . zen_parse_url($url, 'path', true);
+    if ($request_type == 'SSL') { 
+      $catalogHttpsServer = 'https://' . $httpServer;
+      $catalogHttpsUrl = 'https://' . $httpServer . '/' . zen_parse_url($url, 'path', true);
+    } else {
+      $catalogHttpsServer = 'http://' . $httpServer;
+      $catalogHttpsUrl = 'http://' . $httpServer . '/' . zen_parse_url($url, 'path', true);
+    }
```
Comment (drbyte) [inline zc_install/includes/functions/general.php]: In the `else` can we just set them to the non-https variable?  This helps it be slightly clearer about the intent (think semantic), and eliminates a bit of duplication.

ie: 
```php
} else {
  $https = $http;
  $httpsUrl = $httpUrl;
}
```

**Comment by scottcwilson** [comment]
That logic doesn't work; it gives the default text in the https locations.

**Comment by drbyte** [comment]
I don't understand.

**Comment by drbyte** [comment]
Why can't lines 206-207 simply point to the values set in 200-201?

**Comment by scottcwilson** [comment]
screen shot 2016-06-19 at 6 08 40 pm

It doesn't fill in the line.  Oh, unless what you were writing was pseudocode and you meant 

```
    } else {
      $catalogHttpsServer = $catalogHttpServer;
      $catalogHttpsUrl = $catalogHttpUrl;
```

**Comment by drbyte** [comment]
Yes, I was just giving an example, instead of writing the code for you ;)

**Comment by scottcwilson** [comment]
LOL sorry about that, chief.  I took you too literally. :) 
Done in a minute.

**Comment by drbyte** [comment]
No worries. Sometimes it's tough to know how specific to get ;)

**Comment by drbyte** [comment]
👍 
Looks good. 
Is this now working the way you expected on your localhost? ie: solving the problem you were originally experiencing?

**Comment by scottcwilson** [comment]
Works like a dream!  
screen shot 2016-06-19 at 6 37 19 pm

**Comment by zcwilt** [comment]
you could simplify code even further and remove the need for the else logic 

e.g. 

     $catalogHttpsServer = $catalogHttpServer;
     $catalogHttpsUrl = $catalogHttpUrl;
     if ($request_type == 'SSL') { 
       $catalogHttpsServer = 'https://' . $httpServer;
       $catalogHttpsUrl = 'https://' . $httpServer . '/' . zen_parse_url($url, 'path', true);
     }

**Comment by scottcwilson** [comment]
Ian: I think the format that was used and merged is actually better because then you can see at a glance that something untoward is happening. In fact, in a future refinement, I'd like to add a message there just in case the person isn't aware that this is a problem.

**Comment by drbyte** [comment]
I’m not sure I agree.

But, that aside, what sort of message are you contemplating?

**Comment by scottcwilson** [comment]
"HEY, we're using http because your site seems not to support https.  This is not best practice, see <link>." 

or something to that effect.

**Comment by drbyte** [comment]
k.

We already display this when zc_install is run over http instead of https and when it can’t detect working ssl to redirect to automatically:

```php
define('TEXT_ERROR_HTTPS', 'PRO TIP: If possible you should already have installed an SSL certificate, and run the installer using https://');
```

**Pull request merged**